inc/az_ulib_ustream_forward_base.h

```diff
@@ -0,0 +1,374 @@
+     || (handle->control_block->api != &type_api)))
+
+/**
+ * @brief Copy data directly from source (provider) to destination (consumer)
+ * 
+ *  The `az_ulib_ustream_forward_flush` API will copy ALL of the contents of the data from the
+ *    source (provider) directly to the destination buffer provided by the consumer in the 
+ *    `#az_ulib_flush_callback* push_callback`.
+ *  
+ *  In the event that the source (provider) memory is not immediately available without a call
+ *    to an external API (e.g. an HTTP connection to a blob storage provider), the flush operation
+ *    will call this external API (e.g. http_response_body_get) in a loop, invoking the
+ *    `push_callback` each pass of the loop.
+ *  
+ *  The `az_ulib_ustream_forward_flush` API will provide the consumer a pointer to the memory to be
+ *    copied in the form of a `const az_span* const` buffer, to be consumed in the `push_callback`
+ *    function implemented by the consumer. 
+ * 
+ * @note: It is the consumer's responsibility to increment any write offset referenced in the
+ *        context inside the push_callback implementation.
+ * 
+ *  The `az_ulib_ustream_forward_flush` API shall meet the following minimum requirements:
+ *      - The flush operation create a `const az_span* const` from the `Data Source` and hand it
+ *        off to the `push_callback`
+ *      - If the totality of `Data Source` is not available in the provided memory location,
+ *        the API shall request the next portion of the data using the appropriate external API. 
+ *        This process shall be repeated in a loop until the totality of the `Data Source` has been
+ *        copied.
+ *      - If there is no more data to return, the flush shall return #AZ_ULIB_EOF and an empty
+ *        `const az_span* const` shall be created and passed to the `push_callback`.
+ * 
+ * @param[in]       ustream_forward_instance    The #az_ulib_ustream_forward* with the interface of
+ *                                              the ustream_forward.
+ * @param[out]      push_callback               The #az_ulib_flush_callback* for the consumer to handle
+ *                                              the incoming data.
+ * @param[out]      push_callback_context       The #az_context* contract between the caller and
+ *                                              push_callback.
+ * 
+ * @pre     \p ustream_forward_instance shall not be `NULL`.
+ * @pre     \p ustream_forward_instance shall be a valid ustream that is the implemented ustream
+ *             type.
+ * @pre     \p push_callback shall not be `NULL`.
+ * @pre     \p push_callback_context shall not be `NULL`.
+ * 
+ * @return The #az_result with the result of the flush operation
+ *      @retval #AZ_OK                        If the flush operation succeeded in creating the
+ *                                            `const az_span* const` buffer and calling the
+ *                                            `push_callback`.
```

Comment:
Considere something like:
If the flush operation succeeded in pushing the entire stream content to the `push_callback`.